v-next/hardhat-chai-matchers/src/internal/hook-handlers/network.ts

```diff
@@ -0,0 +1,25 @@
+import type {
+  HookContext,
+  NetworkHooks,
+} from "@ignored/hardhat-vnext/types/hooks";
+import type {
+  ChainType,
+  NetworkConnection,
+} from "@ignored/hardhat-vnext/types/network";
+
+import { addChaiMatchers } from "../add-chai-matchers.js";
+
+export default async (): Promise<Partial<NetworkHooks>> => {
+  const handlers: Partial<NetworkHooks> = {
+    async newConnection<ChainTypeT extends ChainType | string>(
+      context: HookContext,
+      next: (context: HookContext) => Promise<NetworkConnection<ChainTypeT>>,
+    ) {
+      addChaiMatchers();
```

Comment:
This should have some mechanism to ensure that `addChaiMatchers` gets called just once.

Now it's called once per network connection, and it's unclear what chai will do

Reply:
Maybe chai allows you to detect if `use` was already called.

Reply:
What about something like [this](https://github.com/NomicFoundation/hardhat/pull/5946/commits/af84d41471f2e84be9f6a565eb5a22bce4790013)?